packages/jest-mock/src/index.ts

```diff
@@ -1003,31 +1022,38 @@ export class ModuleMocker {
     return fn;
   }
 
-  spyOn<T extends {}, M extends NonFunctionPropertyNames<T>>(
-    object: T,
-    methodName: M,
-    accessType: 'get',
-  ): SpyInstance<T[M], []>;
+  spyOn<
+    T extends ClassLike | ObjectLike,
```

Comment:
The `{}` constrain allows any [non-nullish value](https://github.com/typescript-eslint/typescript-eslint/issues/2063#issuecomment-675156492). That’s most like `any`. `string` or `boolean` can be passed and TS does not complain. I was playing with tests. This solution gave the best result.